# Highest paid💰 Athletes🎽 in the world

#### We would seek answers to the questions:

* What was the bigger source of income for athletes in 2020 : salary/prize money or the endorsement fees
* Which sport is more popular based on gross salry & gross endorsement
* Are athelets from individual sports paid more or team sports
* How do women feature on the list over years
* Which country has the most number of players featured on the list over years

## Here is an analysis of the world's Top 10 highest paid athelets from 1990 to 2020. The analysis brings out some interesting facts:

* Tiger Woods dominates the list of Top ranked recently while earlier it was Michael Jordan
* USA dominates the world when it comes to earnings
* Monica Seles is the only women to make the top-10 highest paid athlete list from 1990 to 2020
* Out of Top 10, 3 earners in 2020 are soccer players
* Basketball players earn the most followed by Boxing and Golf.

In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
from datetime import datetime

#Visualisation libraries
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
sns.set()
from plotly.offline import init_notebook_mode, iplot 
import plotly.graph_objs as go
import plotly.offline as py
import pycountry
py.init_notebook_mode(connected=True)
import folium 
from folium import plugins

# Image
import numpy as np
from PIL import Image

# Animation
import matplotlib.ticker as ticker
import matplotlib.animation as animation
from IPython.display import HTML

# Graphics in retina format 
%config InlineBackend.figure_format = 'retina' 

# Increase the default plot size and set the color scheme
plt.rcParams['figure.figsize'] = 8, 5


# Disable warnings in Anaconda
import warnings
warnings.filterwarnings('ignore')
import os


In [2]:
os.listdir()

['.ipynb_checkpoints',
 'Forbes Richest Atheletes (Forbes Richest Athletes 1990-2020).csv',
 'Forbes2020.csv',
 'table_data.csv',
 "The World's Highest-Paid Athletes List.pdf",
 'Untitled.ipynb',
 'Untitled1.ipynb']

### We will explore the data for 1990-2020

In [3]:
df = pd.read_csv(os.listdir()[1])
df.head()

S.NO               Name Nationality  Current Rank Previous Year Rank  \
0     1         Mike Tyson         USA             1                NaN   
1     2     Buster Douglas         USA             2                NaN   
2     3  Sugar Ray Leonard         USA             3                NaN   
3     4       Ayrton Senna      Brazil             4                NaN   
4     5        Alain Prost      France             5                NaN   

         Sport  Year  earnings ($ million)  
0       boxing  1990                  28.6  
1       boxing  1990                  26.0  
2       boxing  1990                  13.0  
3  auto racing  1990                  10.0  
4  auto racing  1990                   9.0

In [4]:
# We will make a copy of the dataframe
df1 = df.copy()
#S.NO is not relevant, so we will drop it from the copy
df1.drop('S.NO', axis=1, inplace=True)
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 301 entries, 0 to 300
Data columns (total 7 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Name                  301 non-null    object 
 1   Nationality           301 non-null    object 
 2   Current Rank          301 non-null    int64  
 3   Previous Year Rank    277 non-null    object 
 4   Sport                 301 non-null    object 
 5   Year                  301 non-null    int64  
 6   earnings ($ million)  301 non-null    float64
dtypes: float64(1), int64(2), object(4)
memory usage: 16.6+ KB


In [6]:
df1.shape

(301, 7)

The Previous Year Rank is null for 24 records, this can be a valid scenario. So we will not impute/drop those

In [7]:
# Convert string to datetime64
df1['Year'] = df1['Year'].apply(pd.to_datetime,format='%Y')

#Set Date column as the index column.
df1['year'] = pd.DatetimeIndex(df1['Year']).year  
df1.set_index('year', inplace=True)
df1.drop('Year',axis=1,inplace=True)

# Converting the sport column to uppercase
df1['Sport'] = df1['Sport'].str.upper()
df1.head()


Name Nationality  Current Rank Previous Year Rank  \
year                                                                   
1990         Mike Tyson         USA             1                NaN   
1990     Buster Douglas         USA             2                NaN   
1990  Sugar Ray Leonard         USA             3                NaN   
1990       Ayrton Senna      Brazil             4                NaN   
1990        Alain Prost      France             5                NaN   

            Sport  earnings ($ million)  
year                                     
1990       BOXING                  28.6  
1990       BOXING                  26.0  
1990       BOXING                  13.0  
1990  AUTO RACING                  10.0  
1990  AUTO RACING                   9.0

### World's Highest-Paid Athletes in 2020

In [10]:
data_2020 = df1[df1.index == 2020]

trace = go.Bar(
                    x = data_2020["earnings ($ million)"],
                    y = data_2020['Name'] ,
                    orientation='h',
                    marker = dict(line=dict(color='black',width=1)),)
data = [trace]
layout = go.Layout(barmode = "group",title="World's Highest-Paid Athletes in 2020",width=800, height=500, 
                       xaxis= dict(title='Earnings in Milion $'),
                       yaxis=dict(autorange="reversed"),
                       showlegend=False)
fig = go.Figure(data = data, layout = layout)
iplot(fig)

### Rank 1 every Year

In [11]:
# Top Paid Athlete for Each Year
Top_paid_each_year = df1[df1['Current Rank'] == 1].sort_values(by='year',ascending=False)

z = Top_paid_each_year[['Name','Sport','Nationality','earnings ($ million)']]

z.style.background_gradient(cmap='Reds')  

Let us find out who featured the most number of times at Rank 1

In [12]:
rank1_count = Top_paid_each_year['Name'].value_counts().to_frame()

In [13]:
rank1_count

Name
Tiger Woods           11
Michael Jordan         6
Floyd Mayweather       4
Michael Schumacher     2
Mike Tyson             2
Cristiano Ronaldo      2
Lionel Messi           1
Roger Federer          1
Evander Holyfield      1

In [14]:
trace = go.Bar(
                    y = rank1_count.index,
                    x = rank1_count['Name'] ,
                    orientation='h',
                    marker = dict(color='pink',
                                 line=dict(color='black',width=1)),
                    )
data = [trace]
layout = go.Layout(barmode = "group",title='Rank 1,maximum number of times',width=800, height=500, 
                       xaxis= dict(title='No of times ranked higest'),
                       yaxis=dict(autorange="reversed"),
                       showlegend=False)
fig = go.Figure(data = data, layout = layout)
iplot(fig)

Tiger Woods featured on the Rank 1 list maximum number of times(11 times) while Messi, Federer and Holyfield achieved the Rank 1 once in the timeframe being observed.

Lets find out how much the Rank 1s earned over years

In [26]:
rank1_gross_earning = Top_paid_each_year.groupby(['Name'])['earnings ($ million)'].sum().to_frame()
rank1_gross_earning = rank1_gross_earning.sort_values(by=['earnings ($ million)'], ascending=False)

In [29]:
def plot_hist(df, x_axis, plot_title, x_label):
    trace = go.Bar(
                    y = df.index,
                    x = df[x_axis] ,
                    orientation='h',
                    marker = dict(line=dict(color='black',width=1)),)
    data = [trace]
    layout = go.Layout(barmode = "group",title=plot_title,width=800, height=500, 
                       xaxis= dict(title=x_label),
                       yaxis=dict(autorange="reversed"),
                       showlegend=False)
    fig = go.Figure(data = data, layout = layout)
    iplot(fig)

In [30]:
plot_hist(rank1_gross_earning, 'earnings ($ million)', 'Rank 1 Gross Earning', 'Rank 1 Gross Earning in Million $')

In [25]:
overall_gross_earnings = df1.groupby(['Name'])['earnings ($ million)'].sum().to_frame().sort_values(by=['earnings ($ million)'], ascending=False)
overall_gross_earnings

earnings ($ million)
Name                                         
Tiger Woods                            1373.8
LeBron James                            844.8
Floyd Mayweather                        840.0
Cristiano Ronaldo                       787.1
Roger Federer                           781.1
...                                       ...
Cecil Fielder                            12.7
Michael Moorer                           12.1
Donovan "Razor" Ruddock                  10.2
Greg Norman                               8.5
Monica Seles                              8.5

[82 rows x 1 columns]

In [31]:
plot_hist(overall_gross_earnings[:10], 'earnings ($ million)', 'TOP 10 Gross Earners', 'Gross Earning in Million $')

Comparing the above two plots, we observe an interesting fact that some athletes did not get on Rank 1 in 20 years but in the list of overall gross earners top 10 list, they feature above the athletes who achieved rank 1 in some year.

Tiger woods retains his rank in the gross earnings at number 1.

### Lets find out how the yearly earnings for Tiger Woods have been varying

In [32]:
def earnings_plot(athlete):
    """
    function that creates a plotly line chart 
    
    """
    athlete_df = df1[df1['Name'] == athlete]

    trace = go.Scatter(
                    x = athlete_df.index,
                    y = athlete_df['earnings ($ million)'] ,
                    orientation='v',
                    marker = dict(color='red',
                                 line=dict(color='red',width=6)),
                    )
    data = [trace]
    layout= go.Layout(title= f'{athlete}' +"'s earnings over the Years",
                  xaxis=dict(title='Year'),
                  yaxis=dict(title="Earnings in US$ (millions)"),)
    fig = go.Figure(data = data, layout = layout)
    iplot(fig)

In [33]:
earnings_plot('Tiger Woods')

### Sport which makes the most number of millionaires

In [34]:
gross_earning_sport = df1.groupby(['Sport'])['earnings ($ million)'].sum().to_frame().sort_values(by=['earnings ($ million)'], ascending=False)
gross_earning_sport

earnings ($ million)
Sport                                             
BASKETBALL                                  3418.1
BOXING                                      2298.8
SOCCER                                      2223.5
GOLF                                        2136.5
TENNIS                                      1017.3
AMERICAN FOOTBALL                            901.2
F1 RACING                                    397.0
AUTO RACING                                  386.3
F1 MOTORSPORTS                               248.0
BASEBALL                                     146.1
MOTORCYCLE GP                                128.0
MMA                                           99.0
NASCAR                                        77.2
ICE HOCKEY                                    57.8
AUTO RACING (NASCAR)                          44.5
NFL                                           37.5
CYCLING                                       28.0
AMERICAN FOOTBALL / BASEBALL                  22.5
HOCKEY                                        17.9
NBA                                           15.2

In [35]:
plot_hist(gross_earning_sport[0:10], 'earnings ($ million)', 'TOP 10 Gross Earnerning Sport', 'Gross Earning in Million $')

In terms of gross earnings, Basketball is the Rank 1. Lets find out the number of occurences of each sport.

In [50]:
millionaires_each_year_per_sport = df1.Sport.value_counts().to_frame()
# millionaires_each_year_per_sport.columns = ['Count']
# millionaires_each_year_per_sport = millionaires_each_year_per_sport.reset_index(level=[0,1])
millionaires_each_year_per_sport

Sport
BASKETBALL                       81
BOXING                           46
GOLF                             44
SOCCER                           33
TENNIS                           23
AUTO RACING                      18
AMERICAN FOOTBALL                17
F1 RACING                         8
BASEBALL                          6
F1 MOTORSPORTS                    5
MOTORCYCLE GP                     4
NFL                               3
ICE HOCKEY                        3
NASCAR                            3
AUTO RACING (NASCAR)              2
NBA                               1
MMA                               1
HOCKEY                            1
CYCLING                           1
AMERICAN FOOTBALL / BASEBALL      1

In [51]:
plot_hist(millionaires_each_year_per_sport[0:10], 'Sport', 'TOP 10 Sport that made most number of millionaires each year', 'Count')

### Countries that feature most number of times

In [52]:
max_sport_country = df1['Nationality'].value_counts().to_frame()
max_sport_country

Nationality
USA                       206
Germany                    13
UK                         13
Switzerland                12
Portugal                   10
Brazil                      9
Argentina                   9
Canada                      6
Italy                       4
France                      3
Finland                     3
Philippines                 3
Northern Ireland            1
Spain                       1
Russia                      1
Austria                     1
Ireland                     1
Serbia                      1
Filipino                    1
Dominican                   1
Mexico                      1
Australia                   1

In [53]:
plot_hist(max_sport_country[0:10], 'Nationality', 'TOP 10 Countries that made most number of millionaires each year', 'Count')

USA dwarfs other countries on the dataset by a huge margin by featuring 206 times on the list.